First PR of a series to update the TCK to current version of spec

gsmet opened the pull request:
Commits should be reviewed one at a time.

 * https://hibernate.atlassian.net/projects/BVTCK/issues/BVTCK-90
 * https://hibernate.atlassian.net/projects/BVTCK/issues/BVTCK-97
 * https://hibernate.atlassian.net/projects/BVTCK/issues/BVTCK-98
 * https://hibernate.atlassian.net/projects/BVTCK/issues/BVTCK-99
 * https://hibernate.atlassian.net/projects/BVTCK/issues/BVTCK-100

gunnarmorling commented on pom.xml:
```diff
@@ -43,28 +43,28 @@
             <name>Emmanuel Bernard</name>
             <email>emmanuel@hibernate.org</email>
-            <organization>JBoss, by Red Hat</organization>
+            <organization>Red Hat, Inc</organization>
```
A dot is missing: "Red Hat, Inc."

gunnarmorling commented on tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/constraintcomposition/ConstraintCompositionTest.java:
```diff
@@ -121,8 +123,7 @@ public void testComposedConstraintsAreRecursive() {
 	@Test
 	@SpecAssertions({
-			@SpecAssertion(section = "3.3", id = "b"),
-			@SpecAssertion(section = "3.4", id = "s")
```
How comes this is gone?

gsmet replied:
Because it does not make sense with what was 3.4. s and I couldn't find an assertion making sense in 3.4.

gunnarmorling commented on tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/constraintcomposition/FrenchAddressListContainer.java:
```diff
@@ -0,0 +1,70 @@
+/*
+* JBoss, Home of Professional Open Source
+* Copyright 2009, Red Hat, Inc. and/or its affiliates, and individual contributors
```
s/2009/2017/ which reminds me of the issue for getting rid of it altogether.

gunnarmorling commented on tests/src/main/java/org/hibernate/beanvalidation/tck/tests/time/ClockProviderTest.java:
```diff
@@ -50,6 +63,8 @@
 @SpecVersion(spec = "beanvalidation", version = "2.0.0")
 public class ClockProviderTest extends Arquillian {
 
+	private static final long ACCEPTABLE_DEVIATION_IN_MS = 10;
```
That one warrants a comment on its purpose.

gunnarmorling commented on tests/src/main/java/org/hibernate/beanvalidation/tck/tests/time/ClockProviderTest.java:
```diff
@@ -148,7 +169,20 @@ public void canConfigureClockProviderForValidator() {
+		Clock systemClock = Clock.systemDefaultZone();
+
+		if ( !systemClock.getZone().equals( clock.getZone() ) ) {
+			throw new RuntimeException( "The default clock provider should use the default system time zone." );
```
RuntimeException? Not AssertionError?

gsmet replied:
At first, they were included in the ClockProvider implementation and it made more sense to use a RuntimeException but you're right, I changed it.

gsmet commented:
@gunnarmorling 
> And shouldn't there be an update to the audit-tck.xml file, too?

You mean like a commit named "Update the tck-audit.xml file"? :) see https://github.com/beanvalidation/beanvalidation-tck/pull/84/commits/399ba537857a287395e435874296d71402ec27c3

gsmet commented:
@gunnarmorling addressed your comments, except for the TCK reference I removed as I didn't find anything making sense and it looked like an error to me.

gunnarmorling commented:
> You mean like a commit ...

Ah yes. Failed prey to this totally useless "feature" of GitHub to collapse random files in the diff view :/
